contracts/src/OPSuccinctDisputeGame.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {OPSuccinctL2OutputOracle} from "./OPSuccinctL2OutputOracle.sol";
+import {CWIA} from "@solady/utils/legacy/CWIA.sol";
+import {LibBytes} from "@solady/utils/LibBytes.sol";
+import {IDisputeGame} from "@optimism/src/dispute/interfaces/IDisputeGame.sol";
+import {Claim, GameStatus, GameType, Hash, Timestamp} from "@optimism/src/dispute/lib/Types.sol";
+import {GameNotInProgress, OutOfOrderResolution} from "@optimism/src/dispute/lib/Errors.sol";
+
+contract OPSuccinctDisputeGame is CWIA, IDisputeGame {
+    using LibBytes for bytes;
+
+    /// @notice The address of the L2 output oracle contract.
+    address internal immutable l2OutpoutOracle;
+
+    /// @notice The timestamp of the game's global creation.
+    Timestamp public createdAt;
+
+    /// @notice The timestamp of the game's global resolution.
+    Timestamp public resolvedAt;
+
+    /// @notice Returns the current status of the game.
+    GameStatus public status;
+
+    constructor(address _l2OutpoutOracle) {
+        l2OutpoutOracle = _l2OutpoutOracle;
+    }
+
+    ////////////////////////////////////////////////////////////
+    //                    IDisputeGame impl                   //
+    ////////////////////////////////////////////////////////////
+
+    function initialize() external payable {
+        // TODO: Set createdAt and status
+
+        (uint256 l2BlockNumber, uint256 l1BlockNumber, bytes memory proof) =
+            abi.decode(extraData(), (uint256, uint256, bytes));
+
+        OPSuccinctL2OutputOracle(l2OutpoutOracle).proposeL2Output(
+            rootClaim().raw(), l2BlockNumber, l1BlockNumber, proof
+        );
+
+        this.resolve();
+    }
+
+    /// @notice Getter for the game type.
+    /// @dev The reference impl should be entirely different depending on the type (fault, validity)
+    ///      i.e. The game type should indicate the security model.
+    /// @return gameType_ The type of proof system being used.
+    function gameType() public view returns (GameType) {
+        // TODO: Retrive and return the game type.
```

Comment:
Does this still need to be added?

Reply:
I set it to `GameTypes.CANNON` for now, and I discussed with OP Labs, they aren't opposed to adding a new `GameTypes.OP_SUCCINCT`

Reply:
What is the use of `GAME_TYPE`? To see the security model? When would `GameTypes.OP_SUCCINCT` be upstreamed?

Reply:
GameTypes are used in DisputeGameFactory to store multiple games in a mapping, with the game type as key.

We don't have a clear view on when GameTypes.OP_SUCCINCT will be upstreamed, but if for instance we add:

```rust
GameType internal constant OP_SUCCINCT = GameType.wrap(3);
```
[here](https://github.com/ethereum-optimism/optimism/blob/3df185a28a2dabd1204a6708c91f7736cbf7ea70/packages/contracts-bedrock/src/dispute/lib/Types.sol#L39), we could temporary do:

```rust
return GameType.wrap(3);
```

Reply:
Sounds good, just add a comment that GameType.wrap(3) will eventually be upstreamed to be OP Succinct. Add a TODO that this `GameType` needs to be changed in the future, and link to the corresponding Optimism contracts PR.